src/test/sync_tests.cpp

```diff
@@ -29,6 +32,38 @@ void TestPotentialDeadLockDetected(MutexType& mutex1, MutexType& mutex2)
     BOOST_CHECK(!error_thrown);
     #endif
 }
+
+#ifdef DEBUG_LOCKORDER
+template <typename MutexType>
+void TestDoubleLock2(MutexType& m)
+{
+    ENTER_CRITICAL_SECTION(m);
+    LEAVE_CRITICAL_SECTION(m);
```

Comment:
Couldn't this use `LOCK(m)`, since `ENTER_CRITICAL_SECTION` is mostly only used internally to the sync module and other code used `LOCK`

Reply:
`LOCK(m)` does not compile if `m` is `std::mutex`:

```
src/sync.h:132:59: error: no type named 'UniqueLock' in 'std::__1::mutex'
template <typename Mutex, typename Base = typename Mutex::UniqueLock>
                                          ~~~~~~~~~~~~~~~~^~~~~~~~~~
```

Reply:
Though, we never use `LOCK` or even `ENTER_CRITICAL_SECTION` on `std::mutex`, so what is the point of testing for that? https://github.com/bitcoin/bitcoin/pull/19337#issuecomment-669211924